crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -799,10 +940,36 @@ async fn validate_warrant_op(
                         )
                         .into());
                     }
-                    action.clone()
+                    (action.clone(), validation_status)
                 };
                 verify_action_signature(action_sig, &action).await?;
 
+                match validation_status {
+                    ValidationStatus::Valid => {
+                        // TODO block the warrant author
+                        return Err(SysValidationError::ValidationOutcome(
+                            ValidationOutcome::InvalidWarrant(
+                                Box::new(op.warrant().clone()),
+                                "Warranted op is valid".into(),
+                            ),
+                        ));
+                    }
+                    ValidationStatus::Abandoned => {
+                        // TODO This isn't really the right way to handle this. If the op is abandoned
+                        //       then this warrant should probably also be abandoned. But there is
+                        //       no implementation of abandon yet.
+                        return Err(SysValidationError::WorkflowError(Box::new(
+                            WorkflowError::Other(Box::new(std::io::Error::other(
+                                "Warranted op is abandoned",
+                            ))),
+                        )));
+                    }
+                    ValidationStatus::Rejected => {
+                        // This is good, the warrant is valid because we also rejected the op
+                        // TODO block the author of the op
```

Comment:
Leaving blocking TODOs as a placeholder. This may not be the right place to actually implement the blocks but still want it tracked approximately.